CONTRIBUTING.md

```diff
@@ -133,6 +133,66 @@ We aim to design the core `kale` modules to be highly **reusable**, generic, and
 - Python IDE: [Visual Studio Code](https://code.visualstudio.com/download), [PyCharm](https://www.jetbrains.com/pycharm/download/)
 - GitHub: [GitHub Desktop](https://desktop.github.com/), [GitHub guides](https://guides.github.com/), [GitHub documentations](https://docs.github.com/en)
 
+### Testing
+
+All new code must be covered by unit tests, and regression tests where appropriate. We will extend test coverage to exisiting code. [General advice on testing should be consulted](https://realpython.com/python-testing/), if needed. This will inform you on how to write tests and help with decisions on what aspects of your code need to be tested. There is some subjectivity involved in deciding how much of the potential behaviour of your code to check.
+
+#### Test runner
+
+`pykale` uses the `pytest` test runner. This offers a balance of functionality, ease of use and wide community support.
+
+#### Unit tests
+
+Within the `tests/` folder is a folder structure that mimics that of the `kale` python module. Unit tests for code in a given file in `kale/` should be placed in their equivalent file in `tests/` e.g. unit tests for a function in `pykale/kale/loaddata/cifar_access.py` should be located in `pykale/tests/loaddata/test_cifar_access.py`.
+
+#### Regression tests
+
+Regression tests should be placed in `tests/regression`. Further subfolders can be added, as required. We plan to add regression tests covering exisiting functionality based on examples in the `examples/` folder.
+
+#### Test data
+
+Data needed for testing should be placed in `tests/data`. This should be limited to small text files e.g. `.csv`, `.json`, `.yml`. Binary data should be stored outside the repository and referenced e.g. using a DOI. Discuss more complex test data requirements for your **pull request** in the motivating **issue**.
+
+#### Common parameters
+
+Consider adding parameters (or objects etc.) that may be useful to multiple tests as fixtures in a [`conftest.py`](
+https://docs.pytest.org/en/stable/fixture.html#conftest-py-sharing-fixtures-across-multiple-files) file, either in `tests/` or the appropriate sub-module.
+
+#### Testing data frames and arrays
```

Comment:
I think `DataFrame` should be one word,  as in Pandas and Spark/PySpark. documentations. 
The same for the following.